base_dav/models/dav_collection.py

```diff
@@ -117,17 +159,35 @@ def to_vobject(self, record):
             )
         return result
 
+    @api.model
+    def _from_vobject_datetime(self, item):
+        if isinstance(item.value, (datetime, date)):
+            return item.value.strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT)
```

Comment:
my guess would be that here the value can carry a timezone, and that you lose in this conversion. You probably need something like

```python
if myval.tzinfo:
    my_val = item.value.astimezone(pytz.utc)
```